lib/rex/proto/mssql/client.rb

```diff
@@ -1,23 +1,70 @@
 require 'metasploit/framework/tcp/client'
 require 'metasploit/framework/mssql/tdssslproxy'
-require 'metasploit/framework/mssql/base'
+require 'rex/proto/mssql/client_mixin'
+require 'rex/text'
+require 'msf/core/exploit'
+require 'msf/core/exploit/remote'
 
-module Metasploit
-  module Framework
+module Rex
+  module Proto
     module MSSQL
-
-      module Client
-        extend ActiveSupport::Concern
+      class Client
         include Metasploit::Framework::Tcp::Client
-        include Metasploit::Framework::MSSQL::Base
-
+        include Rex::Proto::MSSQL::ClientMixin
+        include Rex::Text
+        include Msf::Exploit::Remote::MSSQL_COMMANDS
+        include Msf::Exploit::Remote::Udp
+        include Msf::Exploit::Remote::NTLM::Client
+        include Msf::Exploit::Remote::Kerberos::Ticket::Storage
+        include Msf::Exploit::Remote::Kerberos::ServiceAuthenticator::Options
+
+        attr_accessor :tdsencryption
+        attr_accessor :sock
+        attr_accessor :auth
+        attr_accessor :ssl
+        attr_accessor :ssl_version
+        attr_accessor :ssl_verify_mode
+        attr_accessor :ssl_cipher
+        attr_accessor :proxies
+        attr_accessor :connection_timeout
+        attr_accessor :send_lm
+        attr_accessor :send_ntlm
+        attr_accessor :send_spn
+        attr_accessor :use_lmkey
+        attr_accessor :use_ntlm2_session
+        attr_accessor :use_ntlmv2
+        attr_accessor :windows_authentication
+        attr_reader :framework_module
+        attr_reader :framework
```

Comment:
what uses this `framework` attribute? strikes me as a little odd for a client of any kind to require a framework object
potentially the same question for `framework_module` I know it's being used to pull stuff out of the datastore, but is the object itself needed and/or can we just pass an opts hash instead? 
asking out of curiosity more than anything else, not a blocker

Reply:
Primarily, both of these get used in `mssql_login` - open to alternative implementations though!

Reply:
where in mssql_login? I can't see it being accessed through the client, the mssql_login module provides the scanner with the framework and framework_module though

Reply:
Sorry - the `mssql_login` method in `lib/rex/proto/mssql/client.rb`. The two main cases are:
```
framework_module.fail_with(Msf::Exploit::Failure::BadConfig, 'The Mssql::Rhostname option is required when using kerberos authentication.') if @hostname.blank?
            kerberos_authenticator = Msf::Exploit::Remote::Kerberos::ServiceAuthenticator::MSSQL.new(
              host: domain_controller_rhost,
              hostname: @hostname,
              mssql_port: rport,
              proxies: proxies,
              realm: domain_name,
              username: user,
              password: pass,
              framework: framework,
              framework_module: framework_module,
              ticket_storage: Msf::Exploit::Remote::Kerberos::Ticket::Storage::WriteOnly.new(framework: framework, framework_module: framework_module)
            )
```
and in using `print_status`/`print_error`
Should I investigate a way around it?

Reply:
ah I see it now, I didn't realise it was in code you didn't change